OrbitQt/AccessibilityAdapter.h

```diff
@@ -0,0 +1,188 @@
+| TimeGraph +<--+GlAccessibleInterface +<----------------------+A11yAdapter                  |
++-----------+   +----------------------+  |                    +-----------------------------+
+                                          |
+                                          +
+
+The OrbitGlWidgetAccessible is automatically created when an OrbitGlWidget is constructed by
+installing a QT Accessibility Factory (InstallAccessibilityFactories()).
+
+To make sure adapters created by A11yAdapter::GetOrCreateAdapter are deleted when the corresponding
+GlAccessibleInterface is deleted, these interfaces register themselves in the
+GlAccessibleInterfaceRegistry (see OrbitGlAccessibility.h), which in turn allows to register a
+callback on interface deletion. A11yAdapter registers itself for this callback and cleans up
+interfaces accordingly.
+**/
+
+#ifndef ORBIT_QT_GL_ACCESSIBILITY_H_
+#define ORBIT_QT_GL_ACCESSIBILITY_H_
+
+#include <absl/container/flat_hash_map.h>
+#include <absl/container/flat_hash_set.h>
+
+#include <QAccessible>
+#include <QAccessibleWidget>
+#include <QWidget>
+
+#include "OrbitBase/Logging.h"
+#include "OrbitGlAccessibility.h"
+#include "orbitglwidget.h"
+
+namespace orbit_qt {
+
+/*Instances of this act as adapters from QAccessibleInterface to orbit_gl::GlAccessibleInterface and
+ vice versa. Static methods of A11yAdapter provide factory methods and tracking of adapters.
+
+ See file documentation above for more details.
+ */
+class A11yAdapter : public QAccessibleInterface {
+ public:
+  A11yAdapter() = delete;
+  A11yAdapter(A11yAdapter& rhs) = delete;
+  A11yAdapter(A11yAdapter&& rhs) = delete;
+
+  // vvv Overrides QAccessibleInterface vvv
+  bool isValid() const override {
+    bool result = info_ != nullptr;
+    CHECK(!result || interface_map_.find(info_)->second == this);
+    return result;
+  }
+  QObject* object() const override { return &dummy_; }
+  QAccessibleInterface* focusChild() const override { return nullptr; }
+
+  QAccessibleInterface* parent() const override {
+    return GetOrCreateAdapter(info_->AccessibleParent());
+  }
+  QAccessibleInterface* child(int index) const override {
+    return GetOrCreateAdapter(info_->AccessibleChild(index));
+  }
+  int childCount() const override { return info_->AccessibleChildCount(); }
+  int indexOfChild(const QAccessibleInterface* child) const override;
+  [[nodiscard]] QAccessibleInterface* childAt(int x, int y) const override;
+
+  QString text(QAccessible::Text /*t*/) const override { return info_->AccessibleName().c_str(); }
+  void setText(QAccessible::Text /*t*/, const QString& /*text*/) override{};
+  QRect rect() const override;
+  QAccessible::Role role() const override;
+
+  virtual QAccessible::State state() const override {
+    static_assert(sizeof(QAccessible::State) == sizeof(orbit_gl::A11yState));
+    auto state = info_->AccessibleState();
+    return *reinterpret_cast<QAccessible::State*>(&state);
```

Comment:
Yes, thanks for pointing that out! I wasn't happy with the `reinterpret_cast` in this case because of this weird casting to pointer and back, `absl::bit_cast` is actually much nicer here.